fv/bpf_test.go

```diff
@@ -908,6 +908,59 @@ func describeBPFTests(opts ...bpfTestOpt) bool {
 					})
 				})
 
+				Context("Test load balancer service with no backend", func() {
+					var testSvc *v1.Service
+					tgtPort := 8055
+					externalIP := []string{extIP}
+					srcIPRange := []string{}
+					testSvcName := "test-lb-service-extip"
+					var port uint16
+
+					BeforeEach(func() {
+						if testOpts.connTimeEnabled {
+							Skip("FIXME externalClient also does conntime balancing")
+						}
+						externalClient.EnsureBinary("test-connection")
+						externalClient.EnsureBinary("pktgen")
+						externalClient.Exec("ip", "route", "add", extIP, "via", felixes[0].IP)
+						// create a service workload as nil, so that the service has no backend
+						testSvc = k8sCreateLBServiceWithEndPoints(k8sClient, testSvcName, "10.101.0.10", nil, 80, tgtPort,
+							testOpts.protocol, externalIP, srcIPRange)
+						felixes[1].Exec("ip", "route", "add", "local", extIP, "dev", "eth0")
+						felixes[0].Exec("ip", "route", "add", "local", extIP, "dev", "eth0")
+						port = uint16(testSvc.Spec.Ports[0].Port)
+						pol.Spec.Ingress = []api.Rule{
+							{
+								Action: "Allow",
+								Source: api.EntityRule{
+									Nets: []string{
+										externalClient.IP + "/32",
+									},
+								},
+							},
+						}
+						pol = updatePolicy(pol)
+					})
+					It("should have connectivity from workloads[1][0],[1][1] via external IP to workload 0", func() {
+
+						tcpdump := externalClient.AttachTCPDump("any")
+						tcpdump.SetLogEnabled(true)
+						matcher := fmt.Sprintf("ICMP %s udp port %d unreachable", extIP, port)
+						tcpdump.AddMatcher("ICMP", regexp.MustCompile(matcher))
```

Comment:
I still have a slight doubt that we're not testing the end-user behaviour here; what if tcpdump is outputting `ICMP %s udp port %d unreachable bad checksum` or something?